Upgrade to Gradle 8.8

wjglerum opened the pull request:
Update build to use Gradle 8.8 and match Kotlin version with the embedded Kotlin version of Gradle. Also fixes some deprecations and fixes some suggestions from IntelliJ IDEA.

The Google collections are no longer a transitive dependency, thus rewrote those bits in native Kotlin instead, would be a bit overkill to just include that for the build.

Had this work laying around on a local branch and decided to do this step-by-step instead of updating to the latest Gradle version in one go.

wjglerum commented:
@sergej-koscejev could you please trigger a build and review this work?

sergej-koscejev commented:
Multimap is IMO fine in integration-tests, makes the code nicer and this is a dev-only dependency anyway.

sergej-koscejev commented:
TeamCity build [triggered](https://mps.builds.itemis.cloud/buildConfiguration/Mbeddr2_Mbeddr_Gradle_MpsBuildBackends/375799?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true).

wjglerum commented:
> Multimap is IMO fine in integration-tests, makes the code nicer and this is a dev-only dependency anyway.

Yeah makes sense, just reverted it and added google guava to the buildscript dependencies.

Also configure Java 11 via toolchains as the CI runners probably are running on 17 by default now.

sergej-koscejev commented on launcher/src/main/java/de/itemis/mps/gradle/launcher/MpsBackendBuilder.java:
```diff
@@ -89,7 +90,7 @@ public MpsBackendBuilder withJvmVendorSpec(JvmVendorSpec jvmVendorSpec) {
     public MpsBackendBuilder withJavaExecutable(Provider<String> javaExecutable) {
         this.javaLauncher.set(javaExecutable
-                .map(path -> SpecificInstallationToolchainSpec.fromJavaExecutable(objects, path))
+                .map(path -> SpecificInstallationToolchainSpec.fromJavaExecutable(propertyFactory, path))
```
This change will make this method incompatible with 8.12 and below, but apparently no-one cares because nobody complained about it not working with 8.13 either.

For the record, it could be made compatible with a reflection-based approach like this: https://github.com/specificlanguages/mps-gradle-plugin/blob/5dfceb38ae7cfc1e921147f21668ff72fbcfc214/subprojects/jbr-toolchain/src/main/kotlin/com/specificlanguages/jbrtoolchain/ToolchainSpecFactory.kt

but maybe all those `withJvm` and `withJava` methods should be deprecated and removed instead because they don't seem to bring enough value.
